jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/AwsConnectionManager.kt

```diff
@@ -137,49 +131,61 @@ abstract class AwsConnectionManager(private val project: Project) : SimpleModifi
 
     @Synchronized
     private fun changeFieldsAndNotify(fieldUpdateBlock: () -> Unit) {
-        incModificationCount()
-
-        validationJob?.cancel(CancellationException("Newer connection settings chosen"))
         val isInitial = connectionState is ConnectionState.InitializingToolkit
         connectionState = ConnectionState.ValidatingConnection
 
+        // Grab the current state stamp
+        val modificationStamp = this.modificationCount
+
         fieldUpdateBlock()
 
-        validationJob = GlobalScope.launch(Dispatchers.IO) {
-            val credentialsIdentifier = selectedCredentialIdentifier
-            val region = selectedRegion
+        val validateCredentialsResult = validateCredentials(selectedCredentialIdentifier, selectedRegion, isInitial)
+        validationJob.getAndSet(validateCredentialsResult)?.cancel()
+
+        validateCredentialsResult.onSuccess {
+            // Validate we are still operating in the latest view of the world
+            if (modificationStamp == this.modificationCount) {
```

Comment:
Do we increase modification number anywhere else, or is this the only path?

Reply:
I believe only in this code path where we:
1. Start validating and enter validating state
2. Finish validating and write back the final state based on validation